netutil.py

```diff
@@ -92,7 +92,7 @@ def writeResolverFile(configuration, filename):
 
     for iface in configuration:
         settings = configuration[iface]
-        if settings.isStatic4() and settings.dns:
+        if (not settings.isDynamic()) and settings.dns:
             if settings.dns:
                 for server in settings.dns:
                     outfile.write("nameserver %s\n" % server)
```

Comment:
The original code seems strange: why even check `isStatic` and not rely on `settings.dns` being previously set?  That seems to encode a single policy in 2 places.

It would be worth explaining in the commit message what the exact policy is: if one of v4|v6 is static but the other is dynamic, we don't get to select a DNS because we're necessarily happy with what the other dynamically-provided info is?

Reply:
The dynamic settings will override the statically configured.